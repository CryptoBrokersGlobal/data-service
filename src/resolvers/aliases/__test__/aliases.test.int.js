const createResolver = require('../');

const loadConfig = require('../../../loadConfig');
const createDb = require('../../../db');

const ADDRESS = '3PDSJEfqQQ8BNk7QtiwAFPq7SgyAh5kzfBy';
const ALIAS = 'sexy-boys';

const db = createDb(loadConfig());
const resolverOne = createResolver.one({
  db,
  emitEvent: () => () => null,
});
const resolverMany = createResolver.many({
  db,
  emitEvent: () => () => null,
});

describe('Alias resolver for one', () => {
  it('fetches an alias', async () => {
    const x = await resolverOne(ALIAS)
      .run()
      .promise();
    expect(x).toMatchSnapshot();
  });

  it('returns null for non-existing', async () => {
    const x = await resolverOne('__non-existing__')
      .run()
      .promise();
    expect(x).toBe(null);
  });
});

describe('Alias resolver for many', () => {
  it('fetches aliases for address without showBroken', async () => {
    const aliasesList = await resolverMany({
      address: ADDRESS,
    })
      .run()
      .promise();
    expect(aliasesList).toMatchSnapshot();
  });

  it('fetches aliases for address with showBroken', async () => {
    const aliasesList = await resolverMany({
      address: ADDRESS,
      showBroken: true,
    })
      .run()
      .promise();
    expect(aliasesList).toMatchSnapshot();
  });
});
