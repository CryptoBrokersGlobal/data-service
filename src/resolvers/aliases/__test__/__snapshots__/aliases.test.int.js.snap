// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alias resolver for many fetches aliases for address 1`] = `
Object {
  "__type": "list",
  "data": Array [
    Object {
      "__type": "alias",
      "data": anonymous {
        "address": "3PHXcxfQGAf3SgGGyg1Vendj5ZvjZxvC6KM",
        "alias": "jeffrey",
      },
    },
    Object {
      "__type": "alias",
      "data": anonymous {
        "address": "3PHXcxfQGAf3SgGGyg1Vendj5ZvjZxvC6KM",
        "alias": "hans",
      },
    },
    Object {
      "__type": "alias",
      "data": anonymous {
        "address": "3PHXcxfQGAf3SgGGyg1Vendj5ZvjZxvC6KM",
        "alias": "hansch",
      },
    },
    Object {
      "__type": "alias",
      "data": anonymous {
        "address": "3PHXcxfQGAf3SgGGyg1Vendj5ZvjZxvC6KM",
        "alias": "liqio.io",
      },
    },
    Object {
      "__type": "alias",
      "data": anonymous {
        "address": "3PHXcxfQGAf3SgGGyg1Vendj5ZvjZxvC6KM",
        "alias": "trextracker",
      },
    },
    Object {
      "__type": "alias",
      "data": anonymous {
        "address": "3PHXcxfQGAf3SgGGyg1Vendj5ZvjZxvC6KM",
        "alias": "maid",
      },
    },
    Object {
      "__type": "alias",
      "data": anonymous {
        "address": "3PHXcxfQGAf3SgGGyg1Vendj5ZvjZxvC6KM",
        "alias": "jveldt",
      },
    },
    Object {
      "__type": "alias",
      "data": anonymous {
        "address": "3PHXcxfQGAf3SgGGyg1Vendj5ZvjZxvC6KM",
        "alias": "bunq",
      },
    },
    Object {
      "__type": "alias",
      "data": anonymous {
        "address": "3PHXcxfQGAf3SgGGyg1Vendj5ZvjZxvC6KM",
        "alias": "tikkie",
      },
    },
    Object {
      "__type": "alias",
      "data": anonymous {
        "address": "3PHXcxfQGAf3SgGGyg1Vendj5ZvjZxvC6KM",
        "alias": "rijksoverheid",
      },
    },
    Object {
      "__type": "alias",
      "data": anonymous {
        "address": "3PHXcxfQGAf3SgGGyg1Vendj5ZvjZxvC6KM",
        "alias": "burned",
      },
    },
    Object {
      "__type": "alias",
      "data": anonymous {
        "address": "3PHXcxfQGAf3SgGGyg1Vendj5ZvjZxvC6KM",
        "alias": "j.veldt",
      },
    },
    Object {
      "__type": "alias",
      "data": anonymous {
        "address": "3PHXcxfQGAf3SgGGyg1Vendj5ZvjZxvC6KM",
        "alias": "shapeshift.io",
      },
    },
    Object {
      "__type": "alias",
      "data": anonymous {
        "address": "3PHXcxfQGAf3SgGGyg1Vendj5ZvjZxvC6KM",
        "alias": "changelly.com",
      },
    },
    Object {
      "__type": "alias",
      "data": anonymous {
        "address": "3PHXcxfQGAf3SgGGyg1Vendj5ZvjZxvC6KM",
        "alias": "changelly",
      },
    },
  ],
}
`;

exports[`Alias resolver for one fetches an alias 1`] = `
Object {
  "__type": "alias",
  "data": anonymous {
    "address": "3PQqCXNPoTRsHRK2nZyXWENaZwLmxrUZPBi",
    "alias": "sexy-boys",
  },
}
`;
